python/invoke_release/tasks.py

```diff
@@ -1128,14 +1164,31 @@ def branch(_, verbose=False, no_stash=False):
             raise ReleaseExit()
 
         new_branch = major_branch if proceed_instruction == INSTRUCTION_MAJOR else minor_branch
-        _create_branch_from_tag(verbose, branch_version, new_branch)
 
-        push_instruction = _prompt(
-            'Branch {} created. Would you like to go ahead and push it to remote? (y/N):',
-            new_branch,
-        ).lower()
-        if push_instruction and push_instruction == INSTRUCTION_YES:
-            _push_branch(verbose, new_branch)
+        if USE_PULL_REQUEST:
+            if _is_branch_on_remote(verbose, new_branch):
+                _standard_output(
+                    'Branch {branch} exists on remote. Checking out local tracking branch.',
+                    branch=new_branch,
+                )
+                _create_local_tracking_branch(verbose, new_branch)
+            else:
+                _standard_output(
+                    'Branch {branch} does not exist on remote.\n'
+                    'Creating branch, and pushing to remote.',
+                    branch=new_branch,
+                )
+                _create_branch_from_tag(verbose, branch_version, new_branch)
```

Comment:
lines 1176-1182 and 1184-1191 essentially do the same thing but one asks for a user prompt but the other does not. Should we make both to behave the same way?

Reply:
I was endeavoring not to change the existing functionality for the phabricator workflow. If we choose to push the branch unconditionally in the phabricator workflow, the push may fail.

For the pull request workflow, we always want to push a the newly created branch. So I don't think we want them to behave the same because the use cases are different. Thoughts?

Reply:
I am ok with this difference. 

The only way that a push can fail is if the remote branch has commits that the local branch does not have. But this is not a big issue.

Also after we push the branch_version do we switch to `cherry-pick-<target-branch>-<feature-branch-name>` branch so that the dev can call git cherry-pick  immediately after this.

Reply:
I agree, it makes sense to automate creating the branch so the developer can perform cherry-picks and/or apply fixes immediately. Can you provide specific examples of what the hypothetical values for `<target-branch>` and `<feature-branch-name>` would be in
`cherry-pick-<target-branch>-<feature-branch-name>`?

Reply:
We can prompt for a feature branch name. eg: EB-11233.

We are cherry-picking the fix into 0.16.x release-branch.

The resulting branch would become: `cherry-pick-0.16.x-EB-11233`.

Reply:
👍

Reply:
Automated creation of branch for cherry-picks/fixes in b70e1eba8232181a93f338a20a23ded26c44a90c

Please review again when you have a moment.